### PR TITLE
Chore: Move getUserRoles to service and add cache

### PR DESCRIPTION
**Pull request by sampaiodiego**
## Proposed changes (including videos or screenshots)

Moving this code to a service to make it re-usable and also adding a small cache since this info doesn't change very often.

### CODE REVIEW

File: server/services/authorization/service.ts [hunk 1]
```diff
@@ -83,6 +89,38 @@ export class Authorization extends ServiceClass implements IAuthorization {
...
+			sort: {
+				username: 1,
+			},
+			fields: {
+				username: 1,
+				roles: 1,
+			},
+		};
+
+		const users = await this.Users.findUsersInRoles(roleIds, null, options).toArray();
```
Comment (KevLehman) [inline server/services/authorization/service.ts:115]: If there happens to be a role all users have, this query may return the whole list of users (which can be quite big). Any plans to paginate this? 👀

Comment (sampaiodiego) [reply server/services/authorization/service.ts:115]: I see :thinking: I didn't change the previous behavior though.. maybe we should deprecate this method :thinking:

**Review by ggazzo — APPROVED** [review]

File: server/services/authorization/service.ts (hunk 1 above)
Comment (KevLehman) [reply server/services/authorization/service.ts:115]: Yeah 🤔 that would be a good thing

Comment (sampaiodiego) [reply server/services/authorization/service.ts:115]: Agreed, I'll create a task to deprecate this :+1:

**Pull request merged**